decide.go

```diff
@@ -68,6 +77,37 @@ func addNode(node *Node, expressions []string, object string) (*Node, error) {
 	return node, err
 }
 
+func nextExpression(expressions []string, match exp.Expression) (exp.Expression, []string, error) {
+	if len(expressions) == 0 {
+		return nil, nil, nil
+	}
+
+	if match == nil {
+		// No match, use first
+		expression, err := NewExpression(expressions[0])
+		if err != nil {
+			return nil, nil, err
+		}
+		return expression, expressions[1:], nil
+	}
+
+	// See if the expression exists in the list
+	e := make([]string, 0, len(expressions))
+	for i, exprstr := range expressions {
+		expression, err := NewExpression(exprstr)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if expression.String() == match.String() {
+			return expression, append(e, expressions[i+1:]...), nil
+		}
+		e = append(e, exprstr)
```

Comment:
would it make sense to remove the expression that you passed back as the next expression so then you wouldn't need to keep looping over the set of expressions?